### PR TITLE
Update react-native.d.ts

### PR DESCRIPTION
**Pull request by cailenmusselman**
case 1. Add a new type definition.
- [ ] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [ ] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [ ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

static "positions" is missing from DrawerLayoutAndroid.

From DrawerLayout.android.js: 

var DrawerLayoutAndroid = React.createClass({
  statics: {
    positions: DrawerConsts.DrawerPosition,
  }
  ...
}

### CODE REVIEW

**Comment by bgrieder** [comment]
👍 
Thanks

**Comment by TonyPythoneer** [comment]
Good. But If you're available to provide reference link, I think it will be better.

Example:
[drawerPosition](https://github.com/facebook/react-native/blob/bd60d828c5fc9cb066e5f647c87ecd6f70cb63a5/Libraries/Components/DrawerAndroid/DrawerLayoutAndroid.android.js#L100) - This object includes `Left` and `Right` properties.

**Comment by TonyPythoneer** [comment]
@andy-ms , would you review this? thx.

File: react-native/react-native.d.ts [hunk 1]
```diff
@@ -1931,6 +1931,14 @@ declare namespace  __React {
     export interface DrawerLayoutAndroidStatic extends React.ComponentClass<DrawerLayoutAndroidProperties> {
 
         /**
+         * Specifies the side of the screen from which the drawer will slide in.
+         */
+        positions: {
```
Comment (andy-ms) [inline react-native/react-native.d.ts]: Might `{ Left: number } | { Right: number }` be a better type?

Comment (cailenmusselman) [inline react-native/react-native.d.ts]: I'm not really sure what you're suggesting.... 


Traced the values of Left/Right back from: 

[DrawerLayoutAndroid.android.js](https://github.com/facebook/react-native/blob/master/Libraries/Components/DrawerAndroid/DrawerLayoutAndroid.android.js)
var DrawerConsts = UIManager.AndroidDrawerLayout.Constants;

-- [UIManager.js](https://github.com/facebook/react-native/blob/master/Libraries/Utilities/UIManager.js)
const { UIManager } = NativeModules;

---- [NativeModules.js](https://github.com/facebook/react-native/blob/master/Libraries/BatchedBridge/BatchedBridgedModules/NativeModules.js)

From there it looks like it's dynamically loaded from native code:
[ReactDrawerLayoutManager.java](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/views/drawer/ReactDrawerLayoutManager.java)

```
  @Override
  public @Nullable Map getExportedViewConstants() {
    return MapBuilder.of(
        "DrawerPosition",
        MapBuilder.of("Left", Gravity.START, "Right", Gravity.END));
  }
```

Comment (andy-ms) [inline react-native/react-native.d.ts]: You're right. Thanks for investigating.

**Pull request merged**